inception-external-search-elastic/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/elastic/ElasticSearchProvider.java

```diff
@@ -61,64 +63,69 @@
     {
         List<ExternalSearchResult> results = new ArrayList<ExternalSearchResult>();
 
-        RestTemplate restTemplate = new RestTemplate();
-
-        ElasticSearchResult queryResult;
-
-        String remoteUrl = aTraits.getRemoteUrl();
         String indexName = aTraits.getIndexName();
-        String searchPath = aTraits.getSearchPath();
+        String hostUrl = aTraits.getRemoteUrl().replaceFirst("https?://", "")
+                .replaceFirst("www.", "")
+                .split(":")[0];
         
-        // Set headers
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
+        Settings settings = Settings.builder()
+                .put("client.transport.ignore_cluster_name", "true")
+                .build();
 
-        // Set query
-        String body;
+        TransportClient client = new PreBuiltTransportClient(settings);
         try {
-            body = prepareQuery(aTraits, aQuery);
+            client.addTransportAddress(
+                    new TransportAddress(
+                            InetAddress.getByName(hostUrl),
+                            9300));
         }
-        catch (JsonProcessingException e) {
-            log.error("Invalid JSON while building search query");
-            return Collections.emptyList();
+        catch (UnknownHostException e) {
+            System.out.println(e.getMessage());
+            System.exit(1);
         }
 
-        // Set http entity
-        HttpEntity<String> entity = new HttpEntity<String>(body, headers);
-
-        // Prepare search URL
-        String searchUrl = remoteUrl + "/" + indexName + "/" + searchPath;
+        HighlightBuilder highlightBuilder = new HighlightBuilder();
+        HighlightBuilder.Field highlightField =
+                new HighlightBuilder.Field(aTraits.getDefaultField());
+        highlightField.highlighterType("unified");
+        highlightBuilder.field(highlightField);
 
-        // Send post query
-        queryResult = restTemplate.postForObject(searchUrl, entity,
-                ElasticSearchResult.class);
+        SearchResponse response = client.prepareSearch(indexName)
+                .setQuery(QueryBuilders.termQuery(aTraits.getDefaultField(), aQuery))
```

Comment:
It is not obvious to me that this code maintains the ability to do random ranking of query results?

Reply:
Random ranking of query results is now possible.